packages/ui/app/src/api-page/subpackages/SubpackageEndpointsOverview.tsx

```diff
@@ -0,0 +1,39 @@
+import * as FernRegistryApiRead from "@fern-fern/registry-browser/api/resources/api/resources/v1/resources/read";
+import { useCallback } from "react";
+import { joinUrlSlugs } from "../../docs-context/joinUrlSlugs";
+import { useDocsContext } from "../../docs-context/useDocsContext";
+import { EndpointDescriptor } from "./EndpointDescriptor";
+
+export declare namespace SubpackageEndpointsOverview {
+    export interface Props {
+        subpackage: FernRegistryApiRead.ApiDefinitionSubpackage;
+    }
+}
+
+export const SubpackageEndpointsOverview: React.FC<SubpackageEndpointsOverview.Props> = ({ subpackage }) => {
+    const { navigateToPath, resolvedPathFromUrl } = useDocsContext();
+
+    const handleEndpointClick = useCallback(
+        (endpointDef: FernRegistryApiRead.EndpointDefinition) => {
+            if (resolvedPathFromUrl.type !== "apiSubpackage") {
+                return;
+            }
```

Comment:
I like to avoid these kind of "oh no somehow this is happening" if statements where possible - an alternative is you can pass through the slug from the parent component: 

```jsx
// packages/ui/app/src/api-page/subpackages/ApiSubpackage.tsx
<SubpackageEndpointsOverview subpackage={subpackage} slug={slug} /> 
```

Reply:
You're right, that's much better. Pushed the change.